tests/test_samples.rs

```diff
@@ -224,6 +225,15 @@ fn all_the_fields() {
     assert_eq!(path_dep.source, None);
     assert_eq!(path_dep.kind, DependencyKind::Normal);
     assert_eq!(path_dep.req, semver::VersionReq::parse("*").unwrap());
+    if ver >= semver::Version::parse("1.51.0").unwrap() {
+        assert_eq!(
+            path_dep
+                .path
+                .as_ref()
+                .map(|p| p.ends_with("path-dep/Cargo.toml")),
```

Comment:
I don't think the path ends with `Cargo.toml`, you may want something more like this:

```rust
        assert_eq!(
            path_dep.path.as_ref().unwrap().file_name().unwrap().to_str().unwrap(),
            "path-dep",
        );
```

Reply:
Ah, yes, you're right, I forgot to update the path following the change from `manifest_path` to `path` upstream. Will fix shortly!

Reply:
Also, the reason I did not assert the full string was that I believe it will be an absolute path, not a relative path. Is that not the case?

Reply:
It's an absolute path.  The snippet I posted calls `file_name` just to get the last part.  You could also check if it ends with `tests/all/path-dep`.

Reply:
Ah, missed the call to `file_name`. I think `ends_with("path-dep")` is probably true enough to what we want to test, without the potential of generating false-positives if the test directories get moved around. If you'd strongly prefer the `file_name()` or a longer prefix for `ends_with` though, I'd be happy to update the PR.